p2p/pkg/preconfirmation/tracker/tracker.go

```diff
@@ -222,14 +230,29 @@ func (t *Tracker) handleNewL1Block(
 		}
 		startTime := time.Now()
 
+		var commitmentIdx [32]byte
+		copy(commitmentIdx[:], commitment.CommitmentIndex[:])
+
+		bidAmt, ok := new(big.Int).SetString(commitment.Bid.BidAmount, 10)
+		if !ok {
+			t.logger.Error("failed to parse bid amount", "bidAmount", commitment.Bid.BidAmount)
+			continue
+		}
+
+		opts, err := t.optsGetter(ctx)
+		if err != nil {
+			t.logger.Error("failed to get transact opts", "error", err)
+			continue
+		}
+
 		txHash, err := t.preconfContract.OpenCommitment(
-			ctx,
-			commitment.EncryptedPreConfirmation.CommitmentIndex,
-			commitment.PreConfirmation.Bid.BidAmount,
-			commitment.PreConfirmation.Bid.BlockNumber,
+			opts,
+			commitmentIdx,
+			bidAmt.Uint64(),
+			uint64(commitment.PreConfirmation.Bid.BlockNumber),
 			commitment.PreConfirmation.Bid.TxHash,
-			commitment.PreConfirmation.Bid.DecayStartTimestamp,
-			commitment.PreConfirmation.Bid.DecayEndTimestamp,
+			uint64(commitment.PreConfirmation.Bid.DecayStartTimestamp),
```

Comment:
We could change from `int64` to `uint64` in .proto file and avoid redundant conversion